HACKING.adoc

```diff
@@ -190,3 +191,94 @@ You might even be able to get some feedback early which could save you some time
 You can apply that label to a pull request that is complete and ready for review.
 
 Makes triaging easier.
+
+
+== Node package
+
+=== Initial setup
+
+First you must install and configure {uri-nodejs-download}[Node.js] on your machine.
+
+=== Setup a test project
+
+In order to test the Node package, you need to create a test project adjacent to the clone of the `asciidoctor-reveal.js` repository:
+
+ $ mkdir test-project
+ $ cd test-project
+ $ npm init -y
+
+Now, install the dependencies:
+
+ $ npm i --save asciidoctor.js@1.5.5-2
+ $ npm i --save asciidoctor-template.js@1.5.5-2
+ $ npm i --save ../asciidoctor-reveal.js
+
+IMPORTANT: As mentioned above, the `asciidoctor-reveal.js` repository must be cloned next to the `test-projet` directory.
+
+Create an AsciiDoc file:
+
+[source]
+----
+$ cat <<- EOF > presentation.adoc
+= Title Slide
+// depending on your npm version, you might have to try the other 'revealjsdir' value.
+//:revealjsdir: node_modules/reveal.js
+
+== Slide One
+
+* Foo
+* Bar
+* World
+
+== Slide Two
+
+Hello World - Good Bye Cruel World
+
+[NOTE.speaker]
+--
+Actually things aren't that bad
+--
+EOF
+----
+
+Use the Asciidoctor API to convert an AsciiDoc file into a Reveal.js presentation:
+
+[source]
+----
+$ cat <<- 'EOF' > asciidoctor-revealjs.js
+// Load asciidoctor.js + asciidoctor-template.js
+var asciidoctor = require('asciidoctor.js')();
+var Asciidoctor = asciidoctor.Asciidoctor();
+var Opal = asciidoctor.Opal;
+Opal.load('nodejs');
+Opal.load('pathname');
+require('asciidoctor-template.js');
+
+// Convert the document 'presentation.adoc' using Reveal.js backend
+var attributes = 'revealjsdir=node_modules/asciidoctor-reveal.js/node_modules/reveal.js';
+var options = Opal.hash({safe: 'safe',
+                         backend: 'revealjs',
+                         attributes: attributes});
```

Comment:
I agree that a presentation _could_ be targeting several back-end but in 10+ presentations I did in asciidoc there was always something back-end specific. On the other hand, the reveal.js package location issue is real. Yours is not at the same location than mine. What do you propose as a solution to satisfy both location that don't require crippling OS packages? Maybe a check in the javascript shim? I would accept that.